dymos/transcriptions/explicit_shooting/explicit_shooting_continuity_comp.py

```diff
@@ -1,7 +1,12 @@
+import numpy as np
+
+from ..grid_data import GaussLobattoGrid
+from ...utils.misc import get_rate_units
+
 from ..common.continuity_comp import ContinuityCompBase
 
 
-class ExplicitShootingContinuityComp(ContinuityCompBase):
+class DirectShootingContinuityComp(ContinuityCompBase):
```

Comment:
Now the class name doesn't match the file name; might make sense to make consistent here? Or later.

Reply:
Good catch, fixed

Reply:
> I like the symmetry of input_grid and output_grid, but if the input_grid is only used for the controls, what about something like control_grid, or something more specialized than input? My suggestion's not necessarily better than what's there now.

Input grid will also govern the segments used when I implement multiple shooting. We'll propagate the state from the specified value of the beginning of each segment in multiple shooting, and then enforce state continuity with optimizer constraints.

> To your direct question, would you rename input_grid to grid in all cases, or just when the input and output match? I do like the consistency of the same thing being named the same in all cases.

The more I think about this, the more `output_grid` is a special case that's mostly useful when we're using this as a replacement for simulate, and we want really dense output along each segment but don't care about getting derivatives out or enforcing path constraints.

I think the user specifying just `input_grid` is going to be the predominant use-case by a long shot, and therefore we should just call it `grid`.

Just as we can have secondary timeseries objects I considered doing away with output_grid altogether and having a secondary timeseries object for the simulate use-case, but it won't be named `timeseries` (that's the default one) and therefore won't be as easy to compare the results of `simulate` and `solution` cases.

> I'm still just hung up on the idea in general of grids being GaussLobattoGrid and it's something I use for explicit shooting. Like, that would confuse me as a user, because I think of GaussLobatto as being a method to solve something that's not explicit shooting. Is it also known as a grid? Is there a different way to specify a grid that has that spacing, or is this a generally accepted naming convention? Or is that grid really just used in tests and examples and this is a non-issue?

As an alternative, should we allow `grid` to be specified, but if the user chooses to specify things in the same way as the other transcriptions (`num_segments`, `order`, `segment_ends`), then we just assume they want GaussLobatto?
This would make us a little more swappable with the other transcriptions, and to your point I don't know if the distinction in node placement is as important here, since its _only_ defining the node locations and not the overall transcription technique.

> Is there a mathematical/optimization reason to force the use of GaussLobattoGrid or RadauGrid for the inputs? I tried UniformGrid to test it out. Is it just implementation or the suggestion that the user should use one of those two grids only?

UniformGrid is really only for outputs.  If I try to define the controls on a uniform grid, then the interpolated control values and rates are subject to the Runge-phenomenon and that's generally bad news.  The reason the Radau and Lobatto nodes are used in generally in these applications is because they have nice interpolation properties.

> I tried using no input_grid kwarg to the ExplicitShooting call, but I got the following error:

That makes sense, as without either an input_grid or the number of segments it doesn't know what to do.